From 19a14fb0ba650f26a8af7960b45b80c560571fd9 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 22 Oct 2024 18:14:50 +0200
Subject: [PATCH 1453/1695] drivers: stm32_rng: remove registering to
 shared_resource driver

Remove registering of STM32 RNG driver to shared_resource driver. The
latter is deprecated since integration of the firewall framework
and will soon be removed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git edab3d2bb3eb1f07c3f87b6d0107a2c39e647139 drivers: stm32_rng: remove registering to shared_resource driver]
Change-Id: Ife956c9a4f9a237093e0d5d52f2cb70f02d0a63c
Link: https://github.com/OP-TEE/optee_os/commit/afabc70552d9ddbd0ecbe04dd9fa57b49a389a3e
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/413517
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/stm32_rng.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index f7b057291..6858a3108 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -710,16 +710,6 @@ static TEE_Result stm32_rng_probe(const void *fdt, int offs,
 	if (res)
 		goto err;
 
-#if defined(CFG_STM32MP15)
-	/*
-	 * Only STM32MP15 requires a software registering of RNG secure state
-	 */
-	if (!stm32_etzpc_check_ns_access(STM32MP1_ETZPC_RNG1_ID))
-		stm32mp_register_non_secure_periph_iomem(stm32_rng->base.pa);
-	else
-		stm32mp_register_secure_periph_iomem(stm32_rng->base.pa);
-#endif /* defined(CFG_STM32MP15) */
-
 	/* Power management implementation expects both or none are set */
 	assert(stm32_rng->ddata->has_power_optim ==
 	       stm32_rng->ddata->has_cond_reset);
-- 
2.47.2

